### PR TITLE
Adding DEPENDENCIES.md

### PR DESCRIPTION
**Pull request by bnallapeta**
- Added a new file to list all the dependencies used in Pipeline Service along with their versions.
- Updated README docs to refer to the DEPENDENCIES doc.

### CODE REVIEW

File: DEPENDENCIES.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+|-------------------------------|---------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| OpenShift                     | 4.10                                                                                                                                  | Platform                                                                     | Upgrades to next versions 4.11 need to be tested and certified                                                                                                            |
```
Comment (fgiloux) [inline DEPENDENCIES.md:9]: I find "certified" a bit strong of a commitment

File: DEPENDENCIES.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+| OpenShift Pipelines Operator | openshift-pipelines-operator-rh.v1.7.1 |             |              |
+| OpenShift Cert Manager       | openshift-cert-manager.v1.7.1          |             |              |
+| OpenShift GitOps Operator    | openshift-gitops-operator.v1.5.4       |             |              |
+
+### **Tools**
+
+| **Component**      | **Version**       | **Purpose**                                                                 | **Comments**                                                                                                                                                     |
+|--------------------|-------------------|-----------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| oc (OpenShift CLI) | 4.10              | To interact with the cluster                                                | Follows OpenShift version                                                                                                                                        |
```
Comment (fgiloux) [inline DEPENDENCIES.md:35]: we use kubectl most of the time, which comes with oc but still could be listed. What do you think?

**Review by fgiloux — APPROVED** [review]
this looks good to me. I just have a couple of minor comments.

File: DEPENDENCIES.md (hunk 2 above)
Comment (bnallapeta) [reply DEPENDENCIES.md:35]: Good catch. Let me add that.

File: DEPENDENCIES.md (hunk 1 above)
Comment (bnallapeta) [reply DEPENDENCIES.md:9]: Changed it to 'tested and approved'.

**Review by ramessesii2 — APPROVED** [review]

**Review by avinal — APPROVED** [review]
LGTM

**Comment by avinal** [comment]
Well, my comment was late, I guess 😆

**Pull request merged**